### DISCUSSION TITLE
How to build dbeaver with multi language supporting?

### DISCUSSION THREAD

**Discussion by fanyify** [dbeaver/dbeaver#13679] [Q&A] [answered]
I build dbeaver from the source, I have added the babel p2 repository into the pom file. But when I ran dbeaver on a Chinese OS, the preferences menu was still left with some English world. Can someone help me with this issue?

**Comment by ShadelessFox** [comment, marked as answer]
Hello @fanyify,

If only some words are in English, that's because just haven't been translated to Chinese yet. See more information [here](https://github.com/dbeaver/dbeaver/wiki/Localization).

**Reply by fanyify** [reply]
Thanks for the reply, not just a few words, I have tried using babel to update eclipse 4.18 into Chinese version, those menus already show in Chinese, but when I build dbeaver from the source those menus are still shown in English. :(

**Reply by ShadelessFox** [reply]
Updating Babel may not be so trivial, you can either try adding Babel as p2 repo and then specify required bundles in `feature.xml`, or update it directly from DBeaver using <kbd>Install new software</kbd>.

**Reply by fanyify** [reply]
Thanks for the help, I added `babel eclipse nls` feature into `feature.xml` then it works.

**Reply by bianyun1981** [reply]
> Thanks for the help, I added `babel eclipse nls` feature into `feature.xml` then it works.

Hi, could you please tell me how to add `babel eclipse nls` feature into `feature.xml`?

**Reply by fanyify** [reply]
If you using the IDE tool to open feature.xml, then just add it as a dependency plugin.

**Reply by bianyun1981** [reply]
@fanyify Thanks for your reply. 
I have figured it out by reading this documentation (https://dbeaver.com/docs/wiki/Localization/)
